### PR TITLE
[Map-Viewer] Add layer from OGC API service 

### PR DESCRIPTION
**Pull request by ronitjadhav**
### Description

This pull request adds the feature that allows users to add a layer from an OGC API Service to their map. 

**Currently, it only supports adding features!!**

Following changes were made:
- Added a new `add-layer-from-ogc-api` component in `src/app/components/add-layer-from-ogc-api/add-layer-from-ogc-api.component.ts`
- Updated the `map-context.service.ts` and `map-context.model.ts` to add `MapContextLayerOgcapiModel`
- Updated unit tests to make them pass

### Screenshots

image

image

### Quality Assurance Checklist

- [ ] Commit history is devoid of any _merge commits_ and readable to facilitate reviews
- [ ] If **new logic** ⚙️ is introduced: unit tests were added
- [ ] If **new user stories** 🤏 are introduced: E2E tests were added
- [ ] If **new UI components** 🕹️ are introduced: corresponding stories in Storybook were created
- [ ] If **breaking changes** 🪚 are introduced: add the `breaking change` label
- [ ] If **bugs** 🐞 are fixed: add the `backport <release branch>` label
- [ ] The [documentation website](docs) 📚 has received the love it deserves

#756

### CODE REVIEW

File: libs/feature/map/src/lib/add-layer-from-ogc-api/add-layer-from-ogc-api.component.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import { MapFacade } from '../+state/map.facade'
+import {
+  MapContextLayerModel,
+  MapContextLayerTypeEnum,
+} from '../map-context/map-context.model'
+
+@Component({
+  selector: 'gn-ui-add-layer-from-ogc-api',
+  templateUrl: './add-layer-from-ogc-api.component.html',
+  styleUrls: ['./add-layer-from-ogc-api.component.css'],
```
Comment (jahow) [inline libs/feature/map/src/lib/add-layer-from-ogc-api/add-layer-from-ogc-api.component.ts:13]: Do you think we could make this a standalone component? that would push things towards the right direction :)

**Review by jahow — COMMENTED** [review]
This looks really good! I'll try to see if I can fix the issue with ogc-client provoking errors in the data-fetcher tests and then we can merge :) Thank you!

File: libs/feature/map/src/lib/add-layer-from-ogc-api/add-layer-from-ogc-api.component.ts (hunk 1 above)
Comment (ronitjadhav) [reply libs/feature/map/src/lib/add-layer-from-ogc-api/add-layer-from-ogc-api.component.ts:24]: Done !!

**Review by jahow — APPROVED** [review]
I couldn't have the tests work without the ogc-client mocks, there's something going on with ts-jest but I just can't find it. So I think this can be merged :) thanks!

**Pull request merged**